cql3/expr/expression.cc

```diff
@@ -635,43 +611,29 @@ const auto deref = boost::adaptors::transformed([] (const managed_bytes_opt& b)
 value_list get_IN_values(
         const ::shared_ptr<term>& t, const query_options& options, const serialized_compare& comparator,
         sstring_view column_name) {
-    // RHS is prepared differently for different CQL cases.  Cast it dynamically to discern which case this is.
-    if (auto dv = dynamic_pointer_cast<lists::delayed_value>(t)) {
-        // Case `a IN (1,2,3)`.
-        const auto result_range = dv->get_elements()
-                | boost::adaptors::transformed([&] (const ::shared_ptr<term>& t) { return to_managed_bytes_opt(t->bind_and_get(options)); })
-                | non_null | deref;
-        return to_sorted_vector(std::move(result_range), comparator);
-    } else if (auto mkr = dynamic_pointer_cast<lists::marker>(t)) {
-        // Case `a IN ?`.  Collect all list-element values.
-        const auto val = mkr->bind(options);
-        if (val == constants::UNSET_VALUE) {
-            throw exceptions::invalid_request_exception(format("Invalid unset value for column {}", column_name));
-        }
-        statements::request_validations::check_not_null(val, "Invalid null value for column %s", column_name);
-        return to_sorted_vector(static_pointer_cast<lists::value>(val)->get_elements() | non_null | deref, comparator);
+    const constant in_list = evaluate_IN_list(t, options);
+    if (!in_list.is_null_or_unset() && !in_list.type->without_reversed().is_list()) {
+        throw std::logic_error(format("get_IN_values(single-column) on invalid term {}", in_list.type->without_reversed().name()));
```

Comment:
Why is the type name printed here?  The old code printed the term.

Reply:
I remember something that `term` wasn't giving much information because it was something like: `term@0x727834`. Having the type was much more useful.  But I'm not sure about it, I changed it back to `*t`, it's a safe bet.